### PR TITLE
Validate signer to be in validator set (BFT-384)

### PR DESCRIPTION
**Pull request by moshababo**
Add validation for messages sent by replica to have their signer be in the validator set. 

* A new error type variant has been added to the 2 message types.
* Validation has been introduced at the beginning of the processing of each message.
* 2 tests have been added:
    * `replica_prepare_non_validator_signer`
    * `replica_commit_non_validator_signer`

### CODE REVIEW

File: node/actors/bft/src/leader/replica_commit.rs [hunk 1]
```diff
@@ -58,6 +64,13 @@ impl StateMachine {
         let message = signed_message.msg;
         let author = &signed_message.key;
 
+        consensus
+            .validator_set
+            .index(author)
+            .ok_or(Error::NonValidatorSigner {
+                signer: author.clone(),
+            })?;
```
Comment (brunoffranca) [inline node/actors/bft/src/leader/replica_commit.rs:72]: Nit. Add a comment saying what you are checking.

File: node/actors/bft/src/leader/replica_prepare.rs [hunk 2]
```diff
@@ -70,6 +76,13 @@ impl StateMachine {
         let message = signed_message.msg.clone();
         let author = &signed_message.key;
 
+        consensus
+            .validator_set
+            .index(author)
+            .ok_or(Error::NonValidatorSigner {
+                signer: author.clone(),
+            })?;
```
Comment (brunoffranca) [inline node/actors/bft/src/leader/replica_prepare.rs:84]: Nit. Same here.

**Review by brunoffranca — APPROVED** [review]

File: node/actors/bft/src/leader/replica_commit.rs (hunk 1 above)
Comment (moshababo) [reply node/actors/bft/src/leader/replica_commit.rs:73]: Fixed.

File: node/actors/bft/src/leader/replica_prepare.rs (hunk 2 above)
Comment (moshababo) [reply node/actors/bft/src/leader/replica_prepare.rs:85]: Fixed.

File: node/actors/bft/src/leader/tests.rs [hunk 3]
```diff
@@ -254,23 +271,25 @@ async fn replica_prepare_high_qc_of_future_view() {
     );
 }
 
-#[ignore = "fails/unsupported"]
 #[tokio::test]
-async fn replica_prepare_non_validator_signer() {
```
Comment (pompon0) [inline node/actors/bft/src/leader/tests.rs:259]: how is this test different than replica_prepare_non_validator_signer?

File: node/actors/bft/src/leader/replica_prepare.rs [hunk 4]
```diff
@@ -10,6 +10,12 @@ use zksync_consensus_roles::validator;
 /// Errors that can occur when processing a "replica prepare" message.
 #[derive(Debug, thiserror::Error)]
 pub(crate) enum Error {
+    /// Message signer isn't part of the validator set.
+    #[error("Message signer isn't part of the validator set")]
+    NonValidatorSigner {
+        /// Signer of the message.
+        signer: validator::PublicKey,
```
[the remark this answers is not in the document]
Comment (moshababo) [reply node/actors/bft/src/leader/replica_prepare.rs:17]: Fixed.

File: node/actors/bft/src/leader/tests.rs (hunk 3 above)
Comment (moshababo) [reply node/actors/bft/src/leader/tests.rs:277]: Similar to https://github.com/matter-labs/era-consensus/pull/44#discussion_r1414167099. 

Removed.

**Review by pompon0 — APPROVED** [review]

**Pull request merged**